libraries/Microsoft.Bot.Builder.Expressions/CommonRegex.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+using Antlr4.Runtime;
+using Antlr4.Runtime.Misc;
+using Antlr4.Runtime.Tree;
+
+namespace Microsoft.Bot.Builder.Expressions
+{
+    public class CommonRegex
+    {
+        public static Regex CreateRegex(string pattern)
+        {
+            if (string.IsNullOrEmpty(pattern) || !IsCommanRegex(pattern))
+            {
+                throw new ArgumentException("A regular expression parsing error occurred.");
+            }
+
+            // TODO check pattern
+            return new Regex(pattern, RegexOptions.Compiled);
```

Comment:
Given the pattern here you should not bother to compile--you are creating a new one on every evaluation and then throwing it away.  It would be nice to optimize for the case where the regex was a constant, but given the expression architecture, there is no good way to that.

Reply:
By "creating a new one and throwing away",  you mean Expression library which use this CreateRegex functions right? Because in dialog, I expect a RegexRecongizer would cache the regex object returned by this function. 

So to optimize the case of Expression, we probably can use some kind of cache also, and as you mentioned there is no good way to do that today, because Expression itself is once. Anyhow, the opimization if we want to have, at least should be in the caller layer, not this function, right? or are you suggesting we cache some pattern here?

Reply:
Caching doesn't matter for expressions--given the way it is setup every evaluation will create a new regex.

Reply:
A cache in such a low level, will help on Expression to avoid checking\creating it again for repeated pattern, right?

Reply:
if would look like 
CreateRegex(string pattern) {
  if (cache.haskey(pattern)) return Regex; 

  // check
  // create
}